gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/actors/CheckPermissionsActor.java

```diff
@@ -0,0 +1,381 @@
+ * and Things API to retrieve policy data and permissions.
+ * It responds to the sender with a {@link CheckPermissionsResponse} after evaluating the
+ * permission checks.
+ * <p>
+ * The actor forwards requests to the edge command forwarder to retrieve policies and permission
+ * results, handling various entity and policy resources.
+ *
+ * @since 3.7.0
+ */
+final class CheckPermissionsActor extends AbstractActor {
+
+    private final DittoLogger logger = DittoLoggerFactory.getLogger(CheckPermissionsActor.class);
+    private final ActorRef edgeCommandForwarder;
+    private final ActorRef sender;
+    private final Duration defaultAskTimeout;
+
+
+    /**
+     * Constructor to initialize the actor.
+     *
+     * @param edgeCommandForwarder the actor responsible for forwarding commands to edge services.
+     * @param sender the actor reference to the original sender.
+     * @param defaultTimeout the default timeout for async operations.
+     */
+    private CheckPermissionsActor(final ActorRef edgeCommandForwarder, final ActorRef sender,
+            final Duration defaultTimeout) {
+        this.edgeCommandForwarder = edgeCommandForwarder;
+        this.sender = sender;
+        this.defaultAskTimeout = defaultTimeout;
+        getContext().setReceiveTimeout(defaultTimeout);
+    }
+
+    /**
+     * Creates {@link Props} for the {@link CheckPermissionsActor}.
+     *
+     * @param edgeCommandForwarder the actor responsible for forwarding commands to edge services.
+     * @param sender the actor reference to the original sender.
+     * @param defaultTimeout the default timeout for async operations.
+     * @return a {@link Props} object to create the actor.
+     */
+    public static Props props(final ActorRef edgeCommandForwarder, final ActorRef sender,
+            final Duration defaultTimeout) {
+        return Props.create(CheckPermissionsActor.class, edgeCommandForwarder, sender, defaultTimeout);
+    }
+
+    @Override
+    public Receive createReceive() {
+        return ReceiveBuilder.create()
+                .match(CheckPermissions.class, this::handleCheckPermissions)
+                .match(ReceiveTimeout.class, receiveTimeout -> this.handleReceiveTimeout())
+                .matchAny(msg -> logger.warn("Unknown message: <{}>", msg))
+                .build();
+    }
+
+    private void handleReceiveTimeout() {
+        logger.warn("CheckPermissionsActor timed out. Shutting down due to inactivity.");
+        getContext().stop(getSelf());
+    }
+
+    /**
+     * Handles the {@link CheckPermissions} message by initializing the permission results,
+     * grouping permission checks by entity, and performing policy retrievals.
+     *
+     * @param command the {@link CheckPermissions} message to handle.
+     */
+    private void handleCheckPermissions(final CheckPermissions command) {
+        final List<String> permissionOrder = new LinkedList<>(command.getPermissionChecks().keySet());
```

Comment:
@hu-ahmed an `ArrayList<>()` is completely sufficient here as it also maintains the order ..